modules/client/src/client-common/modules/graphql.ts

```diff
@@ -72,4 +74,33 @@ export class GraphqlModule implements IClientGraphQLCore {
     }
     throw new NoNodesAvailableError("graphql");
   }
+
+  public async request({ query, params, name }: {
+    query: string;
+    params: { [key: string]: any };
+    name?: string;
+  }) {
+    let retries = this.clients.length;
```

Comment:
I would throw an error right away if the length is `0`

Reply:
I already do this check on the getClient wich is called two lines later

Reply:
Right, but for this, we have to call the function, enter the async realm, wait for the promise to resolve, etc.

It's much cleaner to fail on the sync side